### PR TITLE
Fix CaltechHPC environment

### PR DESCRIPTION
**Pull request by nilsvu**
- The previous environment had an issue with loading numeric initial data (took ~40min, now takes 16 seconds).
- It also ran BBH evolutions extremely slowly for some reason (2M in 24 hours).
- It was actually built for cascadelake, not skylake.

Kyle and I reinstalled the environment on skylake, so we should be able to run on all available nodes on CaltechHPC now. I confirmed that the IO issues are fixed and a BBH runs at reasonable speed.

### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `new feature` if appropriate.

### CODE REVIEW

**Comment by nilsvu** [comment]
Marking priority because BBH evolutions are essentially broken on CaltechHPC at the moment.

**Comment by knelli2** [comment]
I'm testing this now. I'll report back once I'm confident everything is working smoothly (or as non-roughly as possible)

File: support/Environments/caltech_hpc_gcc_skylake.sh [hunk 1]
```diff
@@ -28,21 +28,32 @@ spectre_run_cmake() {
...
+          -D BLA_VENDOR=OpenBLAS \
           -D CMAKE_BUILD_TYPE=Release \
+          -D BUILD_DOCS=OFF \
+          -D DEBUG_SYMBOLS=OFF \
           -D MEMORY_ALLOCATOR=JEMALLOC \
           -D BUILD_PYTHON_BINDINGS=ON \
           -D MACHINE=CaltechHpcSkylake \
           -D OVERRIDE_ARCH=skylake \
-          -D ENABLE_PARAVIEW=ON \
+          -D ENABLE_PARAVIEW=OFF \
```
Comment (knelli2) [inline support/Environments/caltech_hpc_gcc_skylake.sh]: Were there issues finding paraview? This is really only an issue for the CLI so not an immediate priority.

File: support/Environments/caltech_hpc_gcc_skylake.sh [hunk 2]
```diff
@@ -13,12 +13,12 @@ spectre_setup_modules() {
 
 spectre_load_modules() {
```
Comment (knelli2) [inline support/Environments/caltech_hpc_gcc_skylake.sh]: [random line] Should we remove the `caltech_hpc_gcc_icelake.sh` env now and rename this one to just `caltech_hpc_gcc.sh`? I think that'll be less confusing for users.

**Review by knelli2 — COMMENTED** [review]
Ok I have tested this out and am now confident that this will work on all nodes of CaltechHPC (skylake, cascadelake, icelake).

Note that we had to add the environment variable `FI_PROVIDER=tcp` to the `skylake-2024-04` environment because the mpi module kept trying to use UCX for the fabric but because of #3886, we can't use UCX, so we go with TCP instead.

File: support/Environments/caltech_hpc_gcc_skylake.sh (hunk 1 above)
Comment (nilsvu) [reply support/Environments/caltech_hpc_gcc_skylake.sh]: No issues, I sent a request to the CaltechHPC admins to build ParaView with the same Python etc as the rest of the build. They haven't responded yet.

File: support/Environments/caltech_hpc_gcc_skylake.sh (hunk 2 above)
Comment (nilsvu) [reply support/Environments/caltech_hpc_gcc_skylake.sh]: Yes I think so. Now or later?

Comment (knelli2) [reply support/Environments/caltech_hpc_gcc_skylake.sh]: I'd say now. Just rip the Band-Aid off. Be sure to also change the cluster installation instructions, maybe adding a little description that whatever node you build on, you can only run on that type of node or newer.

Comment (nilsvu) [reply support/Environments/caltech_hpc_gcc_skylake.sh]: ok done

**Review by knelli2 — APPROVED** [review]

**Pull request merged**